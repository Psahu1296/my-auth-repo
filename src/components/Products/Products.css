.main-wrapper {
  display: flex;
  width: 100%;
  flex-wrap: wrap;
  justify-content: center;
  flex-direction: row;
  padding: 40px;
}

.wrapper {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  flex-wrap: wrap;
  border: 1px solid #dedede;
  border-radius: 12px;
  margin: 20px;
  padding: 15px 24px;
  box-shadow: 2px 5px 8px #ececec;
  background: #fff;
  max-width: 430px;
}

.title {
    font-size: 20px;
    font-weight: 700;
    margin-bottom: 15px;
    /* letter-spacing: 18px; */
}

span {
  font-size: 14px;
}

.product-image {
  width: 144px;
  border-radius: 8px;
  margin-top: auto;
  margin-bottom: 15px;
}
