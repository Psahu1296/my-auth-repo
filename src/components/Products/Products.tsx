import "./Products.css";

import React, { useEffect, useState } from "react";

interface ProductParams {
  id: string;
  title: string;
  category: string;
  price: string;
  description: string;
  image: string;
}

const Products = () => {
  const [products, setProducts] = useState<ProductParams[]>([]);

  useEffect(() => {
    fetch("https://fakestoreapi.com/products")
      .then(res => res.json())
      .then(json => setProducts(json));
  });
  return (
    <div className="main-wrapper">
      {products.map((product: ProductParams) => (
        <Product key={product.id} {...product} />
      ))}
    </div>
  );
};

export default Products;

const Product = (params: ProductParams) => {
  return (
    <div data-testid="product-id" className="wrapper">
      <span data-testid="product-title" className="title">
        Title: {params.title}
      </span>
      <span>RS :{params.price}</span>
      <span>Category: {params.category}</span>
      {/* <span >Description: {params.description}</span> */}
      <img className="product-image" src={params.image} alt="product" />
    </div>
  );
};
