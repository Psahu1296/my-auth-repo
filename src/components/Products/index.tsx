import React, { useEffect, useState } from "react";
import "./Products.css";
import { ProductParams } from "@/components/Product/ProductsTypes";
import Product from "@/components/Product";

const Products = () => {
  const [products, setProducts] = useState<ProductParams[]>([]);

  useEffect(() => {
    fetch("https://fakestoreapi.com/products")
      .then(res => res.json())
      .then(json => setProducts(json));
  }, []);

  return (
    <div className="main-wrapper" data-testid="products-page">
      {products.length > 0 ?products.map((product: ProductParams) => (
        <Product key={product.id} {...product} />
      )): <div className="" data-testid="no-product">No Products found</div>}
    </div>
  );
};

export default Products;

