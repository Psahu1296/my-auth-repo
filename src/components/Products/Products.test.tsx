import { render, screen, waitFor } from "@testing-library/react";

import Products from "./Products";

describe("<Products/>", () => {
  it("Api called properly", async () => {
    render(<Products />);
    await waitFor(() => {
      expect(screen.getByTestId("product-id")).toBeInTheDocument();
    });
    expect(screen.getByTestId("products-page")).toMatchSnapshot()
  });
});
