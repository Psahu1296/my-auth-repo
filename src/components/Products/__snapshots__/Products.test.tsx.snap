// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Products/> Api called properly 1`] = `
<div
  class="main-wrapper"
  data-testid="products-page"
>
  <div
    class="wrapper"
    data-testid="product-id"
  >
    <span
      class="title"
      data-testid="product-title"
    >
      Title: 
      my product test
    </span>
    <span>
      RS :
      free
    </span>
    <span>
      Category: 
      test
    </span>
    <img
      alt="product"
      class="product-image"
      src="https://fakestoreapi.com/img/81fPKd-2AYL._AC_SL1500_.jpg"
    />
  </div>
</div>
`;
